src/collector.ts

```diff
@@ -381,11 +354,148 @@ const addSigListeners = (): void => {
     overrideDefaultConfig(env, args)
     console.log('Config reloaded', CONFIG)
   })
+  process.on('SIGINT', async () => {
+    console.log('DETECTED SIGINT SIGNAL')
+    db.close()
+    process.exit(0)
+  })
   console.log('Registerd signal listeners.')
 }
 
-startServer()
+const startCollector = async () => {
+  // Override default configuration with environment variables and arguments
+  overrideDefaultConfig(env, args)
+
+  if (!config.collectorInfo.secretKey || !config.collectorInfo.publicKey) {
+    console.error('Please provide a keypair in collector config')
+    process.exit(1)
+  }
+
+  // Set crypto hash keys from config
+  Crypto.setCryptoHashKey(CONFIG.hashKey)
+
+  // Initialize the database
+  Storage.initializeDB()
+  Storage.addExitListeners(ws)
+
+  // Initialize data log writer if enabled
+  if (CONFIG.dataLogWrite) await initDataLogWriter()
+
+  // Setup collector socket server if enabled
+  if (CONFIG.enableCollectorSocketServer) setupCollectorSocketServer()
+
+  // Add signal listeners
+  addSigListeners()
+
+  // Start RabbitMQ event consumers or connect to distributor based on collector mode
+  if (CONFIG.collectorMode === collectorMode.MQ) {
+    startRMQEventsConsumers()
+  } else {
+    const CONNECT_TO_DISTRIBUTOR_MAX_RETRY = 10
+    let retry = 0
+    while (!connected) {
+      connectToDistributor()
+      retry++
+      await sleep(2000)
+      if (!connected && retry > CONNECT_TO_DISTRIBUTOR_MAX_RETRY) {
+        throw Error('Cannot connect to the distributor!')
+      }
+    }
+  }
+
+  function shutdownCollector() {
+    // Perform any necessary cleanup here
+    db.close() // Close the database connection
+    console.log('Collector shut down complete.')
+    process.exit(1) // Restart the process
+  }
+  try {
+    await initialSync() // Sync initial data
+  } catch (e) {
+    console.error(`Collector process stopped due to error: ${e.message}`)
+    console.log('Attempting fix..')
+
+    // Starts the syncing process from the beginning
+    const status = await startPatching(0)
```

Comment:
this is not correct right? it should start syncing from whatever checkpoint we have in DB.
I also hope we considered syncing last 100 cycles by default on restart @abdulazeem-tk4vr